### PR TITLE
Add testing against Django 1.10+, django-rest-framework 3.4/3.5 & Python 3.6

### LINKED ISSUE #29: Add testing against Django 1.10+ and django-rest-framework 3.4+
Currently the newest versions tested against are Django 1.9, django-rest-framework 3.3 and Python 3.5.

However there are now newer releases available:
* Django 1.10 and 1.11a1
* django-rest-framework 3.4 and 3.5
* Python 3.6

I have a branch locally to add testing for these (Django 1.10's tests are failing, so they'll initially be marked as "allowed to fail"). Once #28 is merged, I'll open a PR.

### PR DESCRIPTION
**Pull request by edmorley**
For now Django 1.10+ has test failures (#30), so is marked as allowed to fail. Later PRs that work to green up newer Django will then simply adjust .travis.yml and documentation/changelog appropriately.

See individual multi-line commit messages for further details.

Fixes #29.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,17 +4,23 @@
 # When updating the envlist, be sure to also update TOX_ENV in .travis.yml
 envlist =
     docs,
-    {py27,py34,py35}-django1.8-drf{3.2,3.3}
-    # Django 1.9 requires DRF >= 3.3
-    {py27,py34,py35}-django1.9-drf3.3
+    py{27,34,35}-django{1.8,1.9,1.10}-drf{3.4,3.5},
+    # Django 1.11+ is only supported as of django-rest-framework 3.5.
+    # Python 3.6 is only supported as of Django 1.11.
```
Comment (kumar303) [inline tox.ini:9]: Python 2 is also supported by Django 1.11 so maybe this comment needs updating? Or perhaps I read it wrong? I see that the tox rule has Python 2 in it.

**Review by kumar303 — APPROVED** [review]
Thanks for updating all the metadata! I was confused by one comment but I'm going to merge it anyway. If it needs fixing, you can do it in a follow-up or just in the next patch.

File: tox.ini (hunk 1 above)
Comment (edmorley) [reply tox.ini:9]: Yeah this comment says "Python 3.6 is only supported as of Django 1.11" (ie "Python 3.6 support was only officially added in Django >= 1.11"), whereas I think you might have read it as "Django 1.11 only supports Python 3.6".

Comment (kumar303) [reply tox.ini:9]: ah, yes, it makes sense now

**Pull request merged**